src/hotspot/share/gc/shared/gcForwarding.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "gc/shared/gcForwarding.hpp"
+#include "memory/memRegion.hpp"
+#include "runtime/globals_extension.hpp"
+
+HeapWord* GCForwarding::_heap_base = nullptr;
+int GCForwarding::_num_low_bits = 0;
+
+void GCForwarding::initialize_flags() {
+  // Nothing to do here, yet. As soon as we have compact
+  // object headers, we will disable the flag when the
+  // heap size exceeds the narrow-encodable address space.
+
+  // size_t max_narrow_heap_size = (size_t(1) << (NUM_LOW_BITS_NARROW - SHIFT)) * HeapWordSize;
+  // if (UseCompactObjectHeaders && MaxHeapSize >= max_narrow_heap_size) {
+  //  FLAG_SET_DEFAULT(UseCompactObjectHeaders, false);
+  // }
+}
```

Comment:
ZGC doesn't use this and even if it did using MaxHeapSize wouldn't work since we have a larger address space than max heap size. Could you move the call to this function out of  `GCArguments::initialize_heap_sizes` and into the GCs that will use `GCForwarding`?

so could you move this out to the GCs that actually uses it?